crypto/rand/sysrand.c

```diff
@@ -88,21 +92,16 @@ int GFp_sysrand_chunk(void *out, size_t requested) {
 #endif
 #endif
 
-
-/* http://man7.org/linux/man-pages/man2/getrandom.2.html: "Calling
- * getrandom() to read /dev/urandom for small values (<= 256) of buflen is
- * the preferred mode of usage." */
-const size_t GFp_sysrand_chunk_max_len = 256;
-
 int GFp_sysrand_chunk(void *out, size_t requested) {
-  assert(requested <= GFp_sysrand_chunk_max_len);
-  if (syscall(SYS_getrandom, out, requested, 0u) < 0) {
-    if (errno == ENOSYS) {
-      return -1;
+  int r = syscall(SYS_getrandom, out, requested, 0u);
```

Comment:
I noticed in the similar BoringSSL patch they are using `long`. And, according to http://man7.org/linux/man-pages/man2/syscall.2.html, `long` is the right type to use. That means that the function return type needs to be `long` and the FFI declaration in the Rust code needs to use `c::long`. I just added `c::long` in 5c4627a8499e3489a466e280bb606fbed6cc3e67.